bwmf/kldiv_loss_test.go

```diff
@@ -25,12 +25,41 @@ func TestEvaluation(t *testing.T) {
 	fmt.Println("loss val is ", loss_val)
 	fmt.Println("gradient is ", g)
 
-	expectedLoss := 5.999997
+	// the loss is: sum of
+	//
+	//   -----------   ---------------------------------
+	//   | 1.0 1.0 |   | 0.0*log(1.0+s) 1.0*log(1.0+s) |
+	//   |---------|   |-------------------------------|
+	//   | 1.0 1.0 | - | 1.0*log(1.0+s) 0.0*log(1.0+s) |
+	//   |---------|   |-------------------------------|
+	//   | 1.0 1.0 |   | 0.5*log(1.0+s) 0.5*log(1.0+s) |
+	//   -----------   ---------------------------------
+	//
+	// which is then approximately 6 - 3.0*log(1.0+s)
+
+	expectedLoss := 6.0
```

Comment:
I am glad to see all the expectedLoss and expectedGrad were correct. Thanks for the documentation.